Photo-Sharing-Sample/PhotoSharing/ViewModels/ConfirmProfileImageViewModel.swift

```diff
@@ -38,52 +38,52 @@ extension ConfirmProfileImageView {
             }
         }
 
-        func updateProfileImage() {
+        func updateProfileImage() async {
             guard var user = dataStoreService.user else { return }
             guard let pngData = selectedImage?.pngFlattened(isOpaque: true) else {
                 return
             }
 
             user.profilePic = "\(user.username)ProfileImage"
-            let storageOperation = storageService.uploadImage(key: user.profilePic, pngData)
 
-            storageOperation.progressPublisher.sink { progress in
-                DispatchQueue.main.async {
+            do {
+                let storageTask = try await storageService.uploadImage(key: user.profilePic, pngData)
+                storageTask.inProcessPublisher.sink { progress in
```

Comment:
Since async/await is already being used, can't this be updated to use that over Combine?
```swift
Task {
    for await progress in await uploadTask.progress {
        self.progress = progress
    }
}

Also, do we need to return to the main thread here? Maybe adding `MainActor.run` or `receive(on: DispatchQueue.main` if we're sticking with Combine

Reply:
- I would like to keep Combine just to show code sample that uses Combine as well in this sample app. 
- Will update the UI methods at the end in another PR. I am doing this as part by part to address moving to v2 Amplify-Swift.

Reply:
I changed all except one place to have combine.